cuqi/experimental/mcmc/_rto.py

```diff
@@ -111,10 +117,17 @@ def M(x, flag):
             self.M = M  
         else:
             raise TypeError("All likelihoods need to be callable or none need to be callable.")
+        # compute the map
+        sim = CGLS(self.M, self.b_tild, self.current_point, self.maxit, self.tol)            
+        self.map, _ = sim.solve()
```

Comment:
I suggest making map a hidden variable `self._map` because it is not standard in our samplers and we have a custom approach to compute the map for now (through Bayesian problem)